ruby_event_store/spec/mappers/protobuf_spec.rb

```diff
@@ -150,50 +151,72 @@ module RubyEventStore
     specify 'defaults' do
       event = RubyEventStore::Proto.new(data: "One")
       expect(event.event_id).to match(/[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}/)
-      expect(event.metadata).to eq({})
+      expect(event.metadata.to_h).to eq({})
       expect(event.data).to eq("One")
     end
+
+    specify 'metadata' do
+      event = RubyEventStore::Proto.new(data: nil, metadata: {one: 1})
+      expect(event.metadata[:one]).to eq(1)
+      expect do
+        event.metadata['doh']
+      end.to raise_error(ArgumentError)
+    end
   end
 
   module Mappers
     RSpec.describe Protobuf do
       let(:event_id)     { "f90b8848-e478-47fe-9b4a-9f2a1d53622b" }
+      let(:metadata) { {
+        one: 1,
+        two: 2.0,
+        three: true,
+        # four: Date.new(2018, 4, 17),
+        five: "five",
+        # six: Time.utc(2018, 12, 13, 11 ),
+        seven: true,
+        eight: false,
+      } }
+      let(:data) do
+        ResTesting::OrderCreated.new(
+          customer_id: 123,
+          order_id: "K3THNX9",
+        )
+      end
       let(:domain_event) do
         RubyEventStore::Proto.new(
           event_id: "f90b8848-e478-47fe-9b4a-9f2a1d53622b",
-          data: ResTesting::OrderCreated.new(
-            customer_id: 123,
-            order_id: "K3THNX9",
-          ),
-          metadata: {
-            time: Time.utc(2018, 12, 13, 11 ),
-          }
+          data: data,
+          metadata: metadata,
         )
       end
-      let(:proto) { "\n\aK3THNX9\x10{" }
+
+      specify "initialize requires protobuf_nested_struct" do
+        p = Protobuf.allocate
+        def p.require(_name)
+          raise LoadError
+        end
+        expect do
+          p.send(:initialize)
+        end.to raise_error(LoadError, "cannot load such file -- protobuf_nested_struct. Add protobuf_nested_struct gem to Gemfile")
```

Comment:
Seems like `LoadError` isn't captured with the `expect` block but it fails the CI currently

Reply:
This works fine. `rails_event_store` specs failed because `Gemfile` did not contain `protobuf_nested_struct`. Easy fix.